akka-stream-tests/src/test/scala/akka/stream/io/TlsSpec.scala

```diff
@@ -364,7 +389,8 @@ class TlsSpec extends StreamSpec("akka.loglevel=INFO\nakka.actor.debug.receive=o
             .scan(ByteString.empty)(_ ++ _)
             .via(new Timeout(6.seconds))
             .dropWhile(_.size < scenario.output.size)
-            .runWith(Sink.head)
+            .map(bs => {ks.shutdown(); bs})
```

Comment:
To verify that Tls closes connection i switched .runWith(Sink.head) to .runWith(Sink.last). If Tls does not close (it is the current behavior without the fix) the test case will be completed with timeout. Unfortunately all other communication patterns hang on this type of sink since Source completion signal is not propagating thought the decorated flow (i think it could be an another issue related to #17122 but i can be wrong here). ks helps me to complete the stream in the correct place: after the scenario.flow for all communication patterns and between server and client for the patterns with truncation. If you think that detecting leaking stages in this test cases has more value i can switch to my initial testing strategy by implementing a separate test case for this issue and keeping other scenarios unattached. Or  i'm open to other ideas how to keep other scenarios and implement new ones within the same test method.